flow-nodes/api-builder-plugin-format-date/test/test.js

```diff
@@ -167,34 +258,305 @@ describe('flow-node format-date', () => {
+
+			const offsetVal = '+03:00';
+			const { value, output } = await flowNode.formatDate({
+				date: '',
+				format: '',
+				offset: offsetVal
+			});
+
+			expect(output).to.equal('next');
+			expect(value).to.equal(moment().utcOffset(offsetVal).format());
+		});
+
+		it('should apply successfully a valid format on a valid date ahead of valid offset if all parameters enabled and valid', async () => {
+
+			const testDate = '2020-10-23T21:18:30+01:00';
+			const offsetVal = '+03:00';
+			const { value, output } = await flowNode.formatDate({
+				date: testDate,
+				format: 'LL',
+				offset: offsetVal
+			});
+
+			expect(output).to.equal('next');
+			expect(value).to.equal('October 23, 2020');
+		});
+
+		it('should not apply a valid utc offset to an invalid date', async () => {
+
+			const invalidDate = 'clearly-not-a-date';
+			const offsetVal = '+03:00';
+			const { value, output } = await flowNode.formatDate({
+				date: invalidDate,
+				format: '',
+				offset: offsetVal
+			});
+
+			const expectedMessage = `Invalid Date: '${invalidDate}`;
+			expect(value).to.be.instanceOf(Error)
+				.and.to.have.property('message', expectedMessage);
+		});
+
+		it('should fail to format an invalid date input despite valid format input', async () => {
 
 			const invalidDate = 'clearly-not-a-date';
 			const { value, output } = await flowNode.formatDate({
 				date: invalidDate,
 				format: 'L'
 			});
 
-			const expectedMessage = `Invalid date: '${invalidDate}`;
+			const expectedMessage = `Invalid Date: '${invalidDate}`;
+			expect(value).to.be.instanceOf(Error)
+				.and.to.have.property('message', expectedMessage);
+		});
+
+		it('should fail to format an invalid date input despite invalid format input', async () => {
+
+			const invalidDate = 'clearly-not-a-date';
+			const { value, output } = await flowNode.formatDate({
+				date: invalidDate,
+				format: 'clearly-not-a-format'
+			});
+
+			const expectedMessage = `Invalid Date: '${invalidDate}`;
 			expect(value).to.be.instanceOf(Error)
 				.and.to.have.property('message', expectedMessage);
 		});
+
+		it('should fail to format a valid date input with invalid input format', async () => {
+
+			const invalidFormat = 'clearly-not-a-format';
+			const { value, output } = await flowNode.formatDate({
+				date: '2020-10-23T21:18:30+01:00',
+				format: invalidFormat
+			});
+
+			const expectedMessage = `Invalid Format: '${invalidFormat}`;
+			expect(value).to.be.instanceOf(Error)
+				.and.to.have.property('message', expectedMessage);
+		});
+
+		/**
+		 * Expect these VALID formats (which when applied to a valid date returns a specific
+		 * portion of the valid date input eg. day or month), to turn out as INVALID format
+		 * since applying moment on the resulting formatted date output yields an invalid date
+		 */
+		 it('should FAIL to format a valid date input with these valid formats', async () => {
```

Comment:
These are just passing objects to format, so that's why they're invalid.  Your previous test already checks an invalid format.  There's no need to test these.